### PR TITLE
Mutant Kitchen + Ghost Pepper Revived

### PR DESCRIPTION
**Pull request by Kurfursten**
Doesn't it suck that we have a bunch of mutant plants that are scarcely used in recipes? That means you rarely see them grown except as a curiosity. No demand in the kitchen means little reason to mutate them, usually. Doesn't it also kind of disappoint you that we have so few Asiatic dishes and salads? I've been kicking around this PR for a while, just throwing together a sprite or two at a time, but it still needs a few.
- Adds 30 new recipes, listed below.
- Revives Ghost Chili Peppers back from the dead. They're a source of condensed capsaicin and, in high potencies, neurotoxin. Because, you know, capsaicin really is a neurotoxin in real life.
- Ghost Peppers can be eaten by ghosts! Better hide them from the vengeful dead.
- Honey is a couple recipes, so it now uses defines.
- New type of food: chip basket. It works basically like a paper bin, but for chips. These are tortilla chips, of course, not your potato chips you're familiar with.
- New reagent: dipping sauce. You can get this by cooking queso (requires making cheese), salsa (requires Vox garlic), guacamole (requires mutating ice peppers), or hummus (requires raising bees). You can dip any food in dipping sauce, even weird stuff like donuts, but only once.
- Food.dmi has had a massive overhaul to clean it up and make it more friendly to work with. New files: food_container.dmi, food_custom.dmi and food_condiment.dmi. Deleted some old, depreciated food icons that were low quality and replaced. This new space should make working with the file a bit faster.
- Stealth edit: smoothed the eclair sprite so that it looks better.

Pretty thoroughly tested.

untitled

Completed, with icon. Bolded recipes use an ingredient that previously had 1 or less recipe associated with them.

> **5 Flour + Soybeans + Koibeans + Tin of Beans = Three Bean Burrito**
> **2 Moonflower + Glowberries = Midnight Snack**
> **Moonflower + Green Grapes + Blood Tomato + Golden Apple = Primordial Soup**
> **Moonflower + 3 White Beat = Starry Night Salad**
> **Frost Pepper + Spaghetti + 5u Soysauce = Spicy Cold Noodles**
> **Frost Pepper + Garlic + Cabbage + Vinegar + Soy Sauce = Chinese Cold Salad**
> **Frost Pepper + Salt + Lime Juice = Gaucamole (Dipping sauce)**
> **Tomato + Lime Juice + Garlic = Salsa (Dipping sauce)**
> Pepper + 2 Cheese Wedge + Black Pepper = Queso (Dipping sauce)
> **Soybeans + 10u Lemon Juice + 5u Honey = Hummus (Dipping sauce)**
> **Honey + Orange + Rawchicken + Soy Sauce = Honey Citrus Chicken**
> Chili + Cherry = Pimiento
> **Watermelon + Kudzu Pod + Pimiento + Cheese = Confederate Spirit Salad**
> **Carp Meat + Koibeans + Tomato + 10u Flour = Fish Taco Supreme**
> **Carp Meat + Frost Pepper + Lime Juice = Poisson Cru**
> **Chocolatebar + Ice Cream = Icecream Sandwich**
> Basket of chips: 20 flour, 10 salt, 10 corn oil
> **Orange + Grapes + Banana + Melonslice = Fruit Salad**
> **Rawchicken + Grapes + Boiled Egg + 5u Vinegar = Chicken Salad**
> **Pimiento + Cheese + Cabbage + Koibeans = Mexican Chopped Salad**

### CODE REVIEW

**Comment by carshalash** [comment]
Thanks babe.

File: code/modules/reagents/reagent_containers/food/snacks/grown.dm [hunk 1]
```diff
@@ -671,6 +671,22 @@
 	filling_color = "#66CEED"
 	plantname = "icechili"
 
+/obj/item/weapon/reagent_containers/food/snacks/grown/ghostpepper
+	name = "ghost pepper"
+	desc = "This pepper is hainted. And pretty spicy, too."
```
Comment (SkowronX) [inline code/modules/reagents/reagent_containers/food/snacks/grown.dm:684]: haunted*?

Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/food/snacks/grown.dm:684]: https://en.wiktionary.org/wiki/haint

Comment (SkowronX) [inline code/modules/reagents/reagent_containers/food/snacks/grown.dm:684]: that's weird, fuckin english

Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/food/snacks/grown.dm:684]: It's usually associated with rural/uneducated people. I just think it sounds hilarious.

**Comment by Kurfursten** [comment]
I decided I was unsatisfied with Revenge and Vampire's delight, so those are out. 

Instead, I'm adding recipes for Chicken Salad, Grape Salad, Orzo Salad, Minestrone, Chili Con Carne, Chilaquiles, Quiche, Gazpacho, Bruschetta, Yogurt, Gelatin, and Panna Cotta. So the spritework goes on.

That will mean that this PR will ultimately contain 30 recipes - most of which using woefully underused ingredients (most notably the mutant plants, honey, vinegar, soysauce, garlic, grapes, lime juice, and peppers).

**Comment by Blithering** [comment]
> You can dip any food in dipping sauce, even weird stuff like donuts, but only once.

Damn right. No double-dipping savages on this space station!
Also looks good, hoping this gets merged.

**Comment by SkowronX** [comment]
Needs hummus.

**Comment by Kurfursten** [comment]
I could do Hummus. It's not too hard to edit the base dip icon.

**Comment by ghost** [comment]
> chili + cherries = pimiento

How do you even cook something into a grown vegetable, how does that work?
And why even give it a spanish name? I thought peppers were already a thing you could grow in the US.

I really need to learn more about this whole "adopted spanish" lingo

**Comment by Kurfursten** [comment]
the english name for pimiento is cherry pepper. cherry + pepper = cherry pepper; or to answer your more general question of 'how does that work', I'll remind you that you can cook an icecream sundae in the microwave which is pretty suspect too

anyway I wouldn't expect this to finish any time soon unless other people offer to contribute sprites, I'm so busy now

**Comment by Blithering** [comment]
Dibs on the spooky jam sprite, I'll post something later tonight.

**Comment by Blithering** [comment]
How about this for the haunted jam?
jam
(I am too terrible at computers to make an example gif)
Download jam for jam: http://s000.tinyupload.com/index.php?file_id=17612548057646312520

**Comment by Iamgoofball** [comment]
Allow traitors to double dip

**Comment by Kurfursten** [comment]
@Blithering Looks good, although maybe a bit bigger? Most foods are bigger than you think

@Iamgoofball /vg/ has always been hesitant to add any traitor outing tools, but it's a cute idea

**Comment by Iamgoofball** [comment]
Fuck I didn't consider that

I can see the security staff lining people up for a chip etiquette test

**Comment by Blithering** [comment]
Sorry for the delay, new attempt at larger jam, with improved spooking animation: 
jaaam

http://s000.tinyupload.com/index.php?file_id=48326462974795490650

**Comment by Kurfursten** [comment]
Very spooky 👍

**Comment by charmstruck** [comment]
> Chocolatebar + Ice Cream = Icecream Sandwich

Shouldn't it be Sugar Cookie + Ice Cream = Icecream Sandwich?
Also, Ice Cream doesn't exist beyond ice cream cones from the machine and the Sundae, would you be using one of those?

**Comment by Kurfursten** [comment]
Originally it was Sandwich + Icecream = Icecream Sandwich, which, while funny, was a little too ridiculous even for me. Chocolate bar seems to fit pretty well.

Yes, it uses the ice cream cone.

**Comment by Blithering** [comment]
On a similar note:

> Basket of chips: 20 flour, 10 salt, 10 corn oil

Not sure what type of chips you are thinking of, but wouldn't potatoes make more sense than flour for either type?

**Comment by charmstruck** [comment]
@Kurfursten 

> Chocolate bar seems to fit pretty well.

Is this ice cream sandwich just ice cream between two bars of chocolate? That seems a bit weird to me

**Comment by Kurfursten** [comment]
**Short answer**: the ice cream sandwich sprite we have is bar-shaped and chocolate-colored. It's already in the game so you can view it yourself now!

**Long answer**: Our most basic recipe is the burger which involves only flour, but almost hamburger buns use egg, milk, and water - not to mention seasonings like sugar and salt. It's hard for me to even picture what a hamburger bun without egg would look like. Also the sprite for burger distinctly shows some kind of lettuce even though it's not in the recipe. This is mostly a good thing anyway, since if every recipe was complex it would be rather frustrating to be chef. A great example of this is @JMWTurner 's cinnamon pie which constantly annoys me. It's the only pie that isn't 10flour + [ingredient]. Instead it requires 5 Milk, 10 Sugar, 1 Egg as well.

**Comment by Shadowmech88** [comment]
Personally I like the idea of incredibly complex recipes. Even better if they require multiple steps involving a variety of kitchen equipment.

But then I also never play chef, so maybe I'm not the best person to ask.

**Comment by carshalash** [comment]
As king chef, what matters is the nutriment and availability of ingrediants.

**Comment by Kurfursten** [comment]
You need both for the chef to be a workable role. You need to be able to throw down a dozen emergency fried eggs if you show up to the shift late, but complex recipes are what you work on in your downtime to show off.

But I don't think of icecream sandwich as a very fleshy/complex food.

**Comment by Intigracy** [comment]
confrictu

**Comment by Blithering** [comment]
Here's some sprites to help out, sorry I didn't get these done sooner.
http://s000.tinyupload.com/index.php?file_id=09961951922487481694

**Comment by Kurfursten** [comment]
Hey, those look really good! Especially the ones you did original, the yoghurt and pannacotta.

I'll finish this PR up ASAP

**Comment by ComicIronic** [comment]
Getting a handle on open PR numbers - feel free to reopen when you've made changes.

**Comment by Kurfursten** [comment]
>Github removed the re-open button

This site really went downhill fast

Also, there's literally no reason to close old PRs at all. Go close some 4 year old feature requests if you gotta close something.

**Comment by Shadowmech88** [comment]
>Github removed the re-open button

So there's no way to re-open PRs at all now? You have to just make new ones?

**Comment by Kurfursten** [comment]
Maybe it's only when a collab forcefully closes your PR, I dunno

**Comment by Shadowmech88** [comment]
Must be, I can see the re-open button on a PR that I closed myself.

**Comment by Exxion** [comment]
Yeah, you can only re-open PRs you closed yourself, unless you're a collab.
But ask  any collab to do  it when you're ready

**Comment by Probe1** [comment]
1.  Please do go close a bunch of feature requests from 3 years ago.

2.  Fix up the PR just flat out cut everything you don't have a sprite for.

**Comment by Kurfursten** [comment]
@Probe1 If you tell me where we moved the defines for reagents, I'll finish it today

**Comment by Probe1** [comment]
Fuck if I know fam I'm still scratching my head at how "we" screwed up condiments.

**Comment by ComicIronic** [comment]
They're in the _DEFINES folder, under reagents.dm.

**Comment by Kurfursten** [comment]
Thanks, now I know it's safe to trash my version of setup.dm

**Comment by Kurfursten** [comment]
Update: Where am I right now?

>Have all sprites, but I want to add a little detailing to two of them to make them a little more different.
>Need to go back through food.dmi to remove things I removed previously when I split them off into a new file (e.g.: condiments into food_condiments.dmi)
>Need to do thorough testing of all items
>Fix Ghost Jam to use the modern spook proc that allows you to reference the spooking ghost, so that it can be more easily controlled

This is **all doable by the end of day**.

**Comment by Kurfursten** [comment]
All work is done, conducting testing now, will take a little time since there are so many items to cook, examine, and eat. Once that's done we're good to go. I'll update wiki once merged.

Fully Tested
>Chip basket
>Hummus
>Gauc
>Salsa
>Queso
>Chilaquiles
>Chili Con Carne
>Quiche
>Minestone
>Gazpacho
>Gelatin
>Poisson Cru
>Ghost pepper (all plant related stages working too)
>Haunted Jam (+movement on spooking)
>Starry night salad
>Midnight snack
>Primordial soup
>Spicy cold noodles
>Chinese cold salad
>Confederate spirit salad
>Fruit salad
>Pimiento
>Chicken salad
>Grape salad
>Orzo salad
>Mexican salad
>Icecream sandwich
>Honey citrus chicken
>Fish taco supreme

Aaand finally figured out the last two. I was using the wrong defines for their recipe - MILKCREAM doesn't exist, CREAM does. Fucking defines, I thought the whole point we used them was so that it would throw an error when you tried to add one that doesn't exist.

**Comment by Probe1** [comment]
<3

**Comment by Kurfursten** [comment]
100% tested. It took longer than I thought to test it all, but it was worth it to catch the two uncookables and three sprite errors.

File: code/modules/reagents/reagent_containers/food.dm [hunk 2]
```diff
@@ -13,4 +14,109 @@
...
+	icon_state = "salsa"
+	dip_message = "Arriba arriba!"
+
+/obj/item/weapon/reagent_containers/food/dipping_sauce/hummus
+	name = "hummus"
+	desc = "The name 'hummus' literally means 'chickpeas', but this variant on the dip has gained popularity in New Mecca, on Helion Prime."
+	icon_state = "hummus"
+	dip_message = "God is good!"
+
+/obj/item/weapon/chipbasket
```
Comment (ComicIronic) [inline code/modules/reagents/reagent_containers/food.dm:75]: Not so happy with this implementation - haven't you just rewritten a storage object, like the donut box?

Comment (Kurfursten) [reply code/modules/reagents/reagent_containers/food.dm:75]: No, it's more like the paper bin.

And it's been six months but iirc paper bin code is pretty terrible

**Comment by SkowronX** [comment]
Gimme some ajiaco

**Comment by Kurfursten** [comment]
When you sprite it

**Comment by Kurfursten** [comment]
Aaand Wiki is done

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 3]
```diff
@@ -4374,3 +4373,258 @@
...
+	desc = "Even in space, where a north/south orientation is meaningless, the South will rise again."
+	icon_state = "confederatespirit"
+	bitesize = 2
+	New()
+		..()
+		reagents.add_reagent(NUTRIMENT,8)
+
+/obj/item/weapon/reagent_containers/food/snacks/fishtacosupreme
+	name = "fish taco supreme"
+	desc = "There may be more fish in the sea, but there's only one kind of fish in the stars."
```
Comment (9600bauds) [inline code/modules/reagents/reagent_containers/food/snacks.dm:4474]: I love this

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 4]
```diff
@@ -4374,3 +4373,258 @@
...
+	New()
+		..()
+		reagents.add_reagent(NUTRIMENT,5)
+
+/obj/item/weapon/reagent_containers/food/snacks/midnightsnack
+	name = "midnight snack"
+	desc = "Perfect for those occasions when engineering doesn't set up power."
+	icon_state = "midnightsnack"
+	bitesize = 2
+	New()
```
Comment (Probe1) [inline code/modules/reagents/reagent_containers/food/snacks.dm:4391]: smh pathing

Comment (Kurfursten) [reply code/modules/reagents/reagent_containers/food/snacks.dm:4391]: I take it you're planning to absolute path snacks.dm?

Comment (Probe1) [reply code/modules/reagents/reagent_containers/food/snacks.dm:4391]: https://github.com/d3athrow/vgstation13/pull/12852  did it 4 days ago and believe me it was exactly as tedious as you think it was.

Comment (Kurfursten) [reply code/modules/reagents/reagent_containers/food/snacks.dm:4391]: but muh regex

Comment (Probe1) [reply code/modules/reagents/reagent_containers/food/snacks.dm:4391]: I'm honestly glad i did by hand so I could unify the formatting a bit more.

I mean now that its over.  At the time I was ready to die.

Comment (Intigracy) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Yeah, change these to absolute pathing please.

**Comment by Intigracy** [comment]
Oh wait it's already merged that's why.

**Pull request merged**